libparsec/crates/client/src/workspace/store/resolve_path.rs

```diff
@@ -646,3 +658,106 @@ pub(crate) async fn resolve_path_for_reparenting(
         }
     }
 }
+
+/// Retrieve the path and the confinement point of a given entry ID.
+/// Both information are acquired without locking, so keep in mind that
+/// they could get outdated by the time you use them. This is fine however,
+/// since this is typically used for providing information to the user
+/// (e.g. for file and folder stats)
+pub(crate) async fn retrieve_path_from_id(
+    store: &super::WorkspaceStore,
+    entry_id: VlobID,
+) -> Result<(FsPath, PathConfinementPoint), ResolvePathError> {
+    // Initialize the results
+    let mut parts = Vec::new();
+    let mut confinement = PathConfinementPoint::NotConfined;
+
+    // Initialize the loop state
+    let mut current_entry_id = entry_id;
+    let mut current_parent_id = store
+        .get_manifest(entry_id)
+        .await
+        .map_err(|err| match err {
+            super::GetManifestError::Offline => ResolvePathError::Offline,
+            super::GetManifestError::Stopped => ResolvePathError::Stopped,
+            super::GetManifestError::EntryNotFound => ResolvePathError::EntryNotFound,
+            super::GetManifestError::NoRealmAccess => ResolvePathError::NoRealmAccess,
+            super::GetManifestError::InvalidKeysBundle(err) => {
+                ResolvePathError::InvalidKeysBundle(err)
+            }
+            super::GetManifestError::InvalidCertificate(err) => {
+                ResolvePathError::InvalidCertificate(err)
+            }
+            super::GetManifestError::InvalidManifest(err) => ResolvePathError::InvalidManifest(err),
+            super::GetManifestError::Internal(err) => err.context("cannot retrieve path").into(),
```

Comment:
nit: You could maybe implement `From<GetManifestError> for ResolvePathError` and use it below too. I'm not sure about how to do with the internal variant though.